### ISSUE TITLE
wp-config.php loaded with default values.

### ISSUE THREAD

**Issue by sadityalal** [docker-library/wordpress#659]
Hello,

I tried to launch the wordpress using below yaml files in a kubernetes cluster.
I waited for mysql service to start properly and then I created the wordpress resources. I tried to access the wordpress in browser and there was an error "Error connecting to database". I checked the configuration file wp-config.php and noticed that there were default values and it didn't use the environment variable to sent those values.
Could you please help me with this in case I am doing something wrong.

Thanks,
Saurabh

############################
Configuration files
############################

<details>
<summary> mysql </summary>

```yaml
apiVersion: v1
kind: Service
metadata:
  name: mysqlsvc
  labels:
    name: mysqlsvc
    tier: test
spec:
  selector:
    name: msql-app
    tier: test
  type: ClusterIP
  ports:
  - port: 3306
    targetPort: 3306
    name: mysqlapp
    protocol: TCP
  clusterIP: None
---
apiVersion: v1
kind: Secret
metadata:
  name: mysql
  labels:
    name: mysql
    tier: test
data:
  user: d3B1c2Vy
  password: cmVkaGF0MTIz
  database: d29yZHByZXNz
  hostname: bXlzcWxzdmM=
type: Opaque
---
apiVersion: v1
kind: PersistentVolume
metadata:
  name: mysql-pv
  labels:
    name: mysql-pv
    tier: test
spec:
  capacity:
    storage: 5Gi
  volumeMode: Filesystem
  accessModes:
  - ReadWriteMany
  storageClassName: nfs-storage
  persistentVolumeReclaimPolicy: Recycle
  mountOptions:
  - hard
  - nfsvers=4.0
  nfs:
    path: /datamysql
    server: 192.168.56.100
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: mysql-pvc
  labels:
    name: mysql-pvc
    tier: test
spec:
  resources:
    requests:
      storage: 5Gi
  volumeMode: Filesystem
  accessModes:
  - ReadWriteMany
  storageClassName: nfs-storage
  volumeName: mysql-pv
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: mysql-deploy
  labels:
    name: mysql-deploy
    tier: test
spec:
  selector:
    matchLabels:
      name: mysql-app
      tier: test
  template:
    metadata:
      name: mysql-app
      labels:
        name: mysql-app
        tier: test
    spec:
      volumes:
        - name: mysql-pvc
          persistentVolumeClaim:
            claimName: mysql-pvc
      containers:
      - volumeMounts:
        - name: mysql-pvc
          mountPath: /var/lib/mysql
        name: mysql-app
        image: mysql
        env:
        - name: MYSQL_ROOT_PASSWORD
          valueFrom:
            secretKeyRef:
              key: password
              name: mysql
        - name: MYSQL_PASSWORD
          valueFrom:
            secretKeyRef:
              key: password
              name: mysql
        - name: MYSQL_USER
          valueFrom:
            secretKeyRef:
              key: user
              name: mysql
        - name: MYSQL_DATABASE
          valueFrom:
            secretKeyRef:
              key: database
              name: mysql
        ports:
        - containerPort: 3306
        resources: {}
```
</details>

<details>
<summary> wordpress </summary>

```yaml
apiVersion: v1
kind: Service
metadata:
  name: websvc
  labels:
    name: websvc
    tier: test
spec:
  selector:
    name: web-app
    tier: test
  ports:
  - port: 8080
    targetPort: 80
    protocol: TCP
    nodePort: 32007
  type: NodePort
---
apiVersion: v1
kind: PersistentVolume
metadata:
  name: web-pv
  labels:
    name: web-pv
    tier: test
spec:
  capacity:
    storage: 5Gi
  volumeMode: Filesystem
  accessModes:
  - ReadWriteMany
  storageClassName: slow
  persistentVolumeReclaimPolicy: Recycle
  mountOptions:
  - hard
  - nfsvers=4.0
  nfs:
    path: /data/dataweb
    server: 192.168.56.101
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: web-pvc
  labels:
    name: web-pvc
    tier: test
spec:
  resources:
    requests:
      storage: 5Gi
  volumeMode: Filesystem
  accessModes:
  - ReadWriteMany
  volumeName: web-pv
  storageClassName: slow
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: web-dep
  labels:
    name: web-dep
    tier: test
spec:
  selector:
    matchLabels:
      name: web-app
      tier: test
  template:
    metadata:
      name: web-app
      labels:
        name: web-app
        tier: test
    spec:
      containers:
      - name: wordpress
        image: wordpress:5.6.0-php7.4
        ports:
        - containerPort: 80
        resources: {}
        env:
        - name: WORDPRESS_DB_USER
          valueFrom:
            secretKeyRef:
              key: user
              name: mysql
        - name: WORDPRESS_DB_PASSWORD
          valueFrom:
            secretKeyRef:
              key: password
              name: mysql
        - name: WORDPRESS_DB_NAME
          valueFrom:
            secretKeyRef:
              key: database
              name: mysql
        - name: WORDPRESS_DB_HOSTNAME
          valueFrom:
            secretKeyRef:
              key: hostname
              name: mysql
```
</details>

**Comment by wglambert** [comment]
`WORDPRESS_DB_HOSTNAME` should be `WORDPRESS_DB_HOST`

**Comment by sadityalal** [comment]
Hi @wglambert, First of all thanks for the suggestions, I changed the Env Variable name to WORDPRESS_DB_HOST, D connection is still not good and configuration file is still the default one.
Below are the values of ENV.

root@web-dep-676fcc9479-kzn8s:/var/www/html# env | grep WORD
WORDPRESS_DB_HOST=mysqlsvc
WORDPRESS_DB_PASSWORD=redhat123
WORDPRESS_DB_USER=wpuser
WORDPRESS_DB_NAME=wordpress

Container connections are good, I tried them using curl from both the containers. But configuration file is not updating.
Also, when I changed the value and hard coded them in wp-includes/load.php, connection to db is good.

Could you please suggest me something in this case?

Thanks in advance

**Comment by wglambert** [comment]
Because you're using a volume to persist data the change to the environment variable won't be updated as that's done by the entrypoint on first initialization

**Comment by sadityalal** [comment]
So we need to do that manually in case we are using volumes?

**Comment by wglambert** [comment]
Yeah or delete the volume so it initializes a new database according to the environment variables

**Comment by sadityalal** [comment]
Hi,

Just to inform, I tried to write the files again from scratch and and I noticed that this time it is working fine and database connection is also good with Volumes only in first time. I am not sure what was my mistake other than Env name mistake.

Below are the files:
[test.yaml.txt](https://github.com/docker-library/wordpress/files/7529081/test.yaml.txt)
[testweb.yaml.txt](https://github.com/docker-library/wordpress/files/7529084/testweb.yaml.txt)

But, thanks for answers.
